src/components/apiTokens/CreateAPIToken.tsx

```diff
@@ -0,0 +1,287 @@
+        const noCustomDate = formData.dateType === 'Custom' && !customDate
+        if (!nameValidation.isValid || noCustomDate) {
+            toast.error('Some required fields are missing')
+            setFormDataErrorObj({
+                invalidName: !nameValidation.isValid,
+                invalidaNameMessage: nameValidation.message,
+                invalidCustomDate: noCustomDate,
+            })
+            return
+        }
+
+        setLoader(true)
+
+        try {
+            const payload = {
+                name: formData.name,
+                description: formData.description,
+                expireAtInMs: formData.expireAtInMs,
+            }
+
+            const { result } = await createGeneratedAPIToken(payload)
+
+            if (result) {
+                const userPermissionPayload = createUserPermissionPayload(
+                    result.userId,
+                    result.userIdentifier,
+                    serverMode,
+                    userGroups,
+                    directPermission,
+                    chartPermission,
+                    adminPermission === 'SUPERADMIN',
+                )
+
+                const { result: userPermissionResponse } = await saveUser(userPermissionPayload)
+                if (userPermissionResponse) {
+                    setTokenResponse(result)
+                    setShowGenerateModal(true)
+                }
+            }
+        } catch (error) {
+            showError(error)
+        } finally {
+            setLoader(false)
+        }
+    }
+
+    const handlePermissionType = (e) => {
+        setAdminPermission(e.target.value)
+    }
+
+    const handleDatesChange = (e) => {
+        onChangeFormData(e, 'customDate')
+    }
+
+    return (
+        <>
+            <div className="cn-9 fw-6 fs-16">
+                <span className="cb-5 cursor" onClick={redirectToTokenList}>
+                    API tokens
+                </span>{' '}
+                / New API token
+            </div>
+            <p className="fs-13 fw-4">
+                API tokens function like ordinary OAuth access tokens. They can be used instead of a password for Git
+                over HTTPS, or can be used to authenticate to the API over Basic Authentication.
+            </p>
+
+            <div className="bcn-0 br-8 en-2 bw-1">
+                <div className="p-20">
+                    <div>
+                        <label className="form__row w-400">
+                            <span className="form__label">
+                                Name <span className="cr-5">*</span>
+                            </span>
+                            <input
+                                tabIndex={1}
+                                placeholder="Name"
+                                className="form__input"
+                                value={formData.name}
+                                onChange={(e) => onChangeFormData(e, 'name')}
```

Comment:
we should add the name field on input so that we can avoid inline onchange function

Reply:
Please do the same for every input on this page